pkg/operator/starter.go

```diff
@@ -247,6 +250,35 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 	configInformers.Start(ctx.Done())
 	dynamicInformers.Start(ctx.Done())
 
+	// REMOVE(4.9) Poll every 30 minutes to ensure removal of 4.7 secrets that
+	// were replaced in 4.8 with a unified secret. Removal is not critical,
+	// it's just potentially confusing if a bunch of now unused secrets linger
+	// forever.
+	secretClient := kubeClient.CoreV1().Secrets(operatorclient.TargetNamespace)
+	secretNamePrefixes := []string{"etcd-all-peer", "etcd-all-serving", "etcd-all-serving-metrics"}
+	go wait.Until(func() {
+		// The 4.7 secrets are likely to have multiple revisions
+		// (i.e. <prefix>-[0-9]*) so it's necessary to list all secrets and
+		// match by prefix.
+		secrets, err := secretClient.List(ctx, metav1.ListOptions{})
+		if err != nil {
+			klog.Warningf("Failed to list secrets when searching for 4.7 secrets to remove: %v", err)
+		}
+		if secrets == nil {
+			return
+		}
+		for _, secret := range secrets.Items {
+			for _, prefix := range secretNamePrefixes {
+				if strings.HasPrefix(secret.Name, prefix) {
+					err := secretClient.Delete(ctx, secret.Name, metav1.DeleteOptions{})
```

Comment:
Not sure if we want to do this. Should we mark the revision abandoned or something so pruner just cleans it up? I wonder why keep any assets for a revision if it is not complete.

Reply:
These secrets only aggregate certs sourced from node-specific secrets. In the event of a downgrade, these 'all' secrets would just be recreated by the 4.7 operator. I'm not sure I understand the concern, can you be more specific?

Reply:
Maybe I am just misunderstanding the code here is my concern.

so each revision creates a versioned secret.

```
etcd-all-peer                                                     Opaque                                6      29m
etcd-all-peer-2                                                   Opaque                                6      29m
etcd-all-peer-3                                                   Opaque                                6      21m
etcd-all-serving                                                  Opaque                                6      29m
etcd-all-serving-2                                                Opaque                                6      29m
etcd-all-serving-3                                                Opaque                                6      21m
etcd-all-serving-metrics                                          Opaque                                6      29m
etcd-all-serving-metrics-2                                        Opaque                                6      29m
etcd-all-serving-metrics-3                                        Opaque                                6      21m
```
My concern is going through these old revisions and removing state from them vs having the unused revision completely removed/pruned. Let me reread the code.

Reply:
How is the pruner going to remove revisions for resources it is no longer tracking?